### DISCUSSION TITLE
What are the benefits of adding  `APPS_DIR` to the `PATH`?

### DISCUSSION THREAD

**Discussion by zelenyjan** [cookiecutter/cookiecutter-django#4493] [Q&A]
This code is in `manage.py`, `wsgi.py` etc. Trying to figure what are the benefits. Seems everything works fine w/o it(at least in my case). Any example what won't work without it?

```python
 current_path = Path(__file__).parent.resolve()
 sys.path.append(str(current_path / "{{ cookiecutter.project_slug }}"))
```

Thanks!

**Comment by luzfcb** [comment]
Cookiecutter-django doesn't follow the standard project structure suggested by the Django start project, instead all apps in the project live within a namespace.

The benefits of this approach are:

1 - As the project grows, the number of apps will grow, which would result in many more items in the root directory of the project, making it more confusing about what django apps are, what settings folders are, other items that are not django apps .  By storing all apps within one namespace this problem is alleviated.

2 - Using namespace also makes it easier when you need to extract a Django app that was born in your project and make it a pluggable app installable via pip


That is, the real benefit will be in the long term, in facilitating the project's maintainability

**Reply by browniebroke** [reply]
I would also add another benefit of using a top level namespace, related to the second point: avoid name conflict with 3rd party apps.

**Reply by zelenyjan** [reply]
Thanks! I use this approach. I made this sample [repo](https://github.com/zelenyjan/cookicutter-sample/tree/main) using this cookiecutter template. Then removed adding `APPS_DIR` to the `PATH` in [manage.py](https://github.com/zelenyjan/cookicutter-sample/blob/main/manage.py) and tried `makemigrations`, `migrate` and `runserver` and everything works fine. So can you please provide any example what won't work? :)

**Reply by luzfcb** [reply]
Good point. I really don't remember when or why this specific setting was included.  After some digging, it was included in [this commit](https://github.com/cookiecutter/cookiecutter-django/commit/6453fea417de512a2219da5011e5896f61032d74) made by the original creator of this project
Unfortunately, the original question as to why still persists since the commit was not included via pull-request and there is no explanation beyond the commit message itself

https://github.com/cookiecutter/cookiecutter-django/commits/master?before=7844cfe008941f3079407d66e54cc4854b935fb5+4855&branch=master&qualified_name=refs%2Fheads%2Fmaster
